### PR TITLE
ntopng: 5.2.1 -> 5.6, ndpi: 4.2 -> 4.6

### PR DESCRIPTION
**Pull request by fabaff**
Changelog: https://github.com/ntop/ntopng/blob/5.6/CHANGELOG.md

Diff: https://github.com/ntop/nDPI/compare/refs/tags/4.2...4.6

Changelog: https://github.com/ntop/nDPI/blob/4.6/CHANGELOG.md

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 223938` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>ndpi</li>
    <li>ntopng</li>
  </ul>
</details>

File: pkgs/tools/networking/ntopng/default.nix [hunk 1]
```diff
@@ -4,51 +4,75 @@
...
+  version = "5.6";
 
   src = fetchFromGitHub {
     owner = "ntop";
     repo = "ntopng";
-    rev = version;
-    sha256 = "sha256-FeRERSq8F3HEelUCkA6pgNNcP94xrWy6EbJgk+cEdqc=";
+    rev = "refs/tags/${version}";
+    hash = "sha256-E5/yrCaC406KlBBuy2Cue7PBXaN/kYfCZQyZSCgm9jA=";
+    leaveDotGit = true;
```
Comment (bjornfor) [inline pkgs/tools/networking/ntopng/default.nix:36]: AFAIK, leaveDotGit isn't reproducible, better not use it in nixpkgs.

**Review by bjornfor — COMMENTED** [review]
IMHO it's better to have stand-alone "reformat expression" commits than do refactoring inside "add changelog to meta" commits.

File: pkgs/tools/networking/ntopng/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/tools/networking/ntopng/default.nix]: Thanks, remove as it was a left-over anyways.

**Review by bjornfor — APPROVED** [review]

**Pull request merged**